extensions-core/histogram/src/main/java/org/apache/druid/query/aggregation/histogram/FixedBucketsHistogramBufferAggregator.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation.histogram;
+
+import org.apache.druid.common.config.NullHandling;
+import org.apache.druid.query.aggregation.BufferAggregator;
+import org.apache.druid.query.monomorphicprocessing.RuntimeShapeInspector;
+import org.apache.druid.segment.BaseObjectColumnValueSelector;
+
+import java.nio.ByteBuffer;
+
+public class FixedBucketsHistogramBufferAggregator implements BufferAggregator
```

Comment:
Seems to prone to #3956.

@jon-wei @gianm @clintropolis you seem to be writing new aggregators frequently, could you fix #3956 so that each separate aggregator doesn't need to roll it's own concurrency?

Reply:
I don't feel that we've reached consensus on #3956 about how to approach the problem. Both high level handling (in IncrementalIndex) and low level handling (in aggregators) have been discussed, but I don't feel we've reached a conclusion about how to fix it.

I also don't think #3956 is very urgent to fix for BufferAggregators right now: they're only used by groupBy v1 in `useOffheap` mode, but in groupBy v1, `aggregate` and `get` are not called concurrently. So nothing bad happens. But it would matter more in the future if IncrementalIndex starts using BufferAggregators.

Reply:
I chimed in on #3956 with 2¢.